### PR TITLE
prov/rxm, util: FI_SOURCE optimization and rename srx

### PR DESCRIPTION
**Pull request by aingerson**

### CODE REVIEW

**Comment by shefty** [comment]
Is the branch up to date?  EFA basically failed everywhere with segfaults, which shouldn't be impacted by these changes.

**Comment by aingerson** [comment]
@shefty it was pretty up to date but I rebased just in case. Still failing.
@wzamazon Do you think the AWS failures are related to the FI_SOURCE change or node issues?

**Comment by wzamazon** [comment]
>  Do you think the AWS failures are related to the FI_SOURCE change or node issues?

I think the failure is related to this PR. I will try to reproduce manually

**Comment by wzamazon** [comment]
@aingerson 

Can you add https://github.com/ofiwg/libfabric/pull/8427

to your PR? I think it should fix the EFA issue

File: prov/util/src/util_attr.c [hunk 1]
```diff
@@ -1127,6 +1127,8 @@ static uint64_t ofi_get_caps(uint64_t info_caps, uint64_t hint_caps,
 	} else {
 		caps = (hint_caps & OFI_PRIMARY_CAPS) |
 		       (attr_caps & OFI_SECONDARY_CAPS);
+
+		caps = (caps & ~FI_SOURCE) | (hint_caps & FI_SOURCE);
```
[the remark this answers is not in the document]
Comment (aingerson) [reply prov/util/src/util_attr.c:1131]: The assumption (according to the below comment in the utility code) is that the hints have already been validated against the info/attributes. This function is just altering them to add/remove other capabilities if necessary.
```
/*
 * Alter the returned fi_info based on the user hints.  We assume that
 * the hints have been validated and the starting fi_info is properly
 * configured by the provider.
 */
```
Adding the attributes to the check would be redundant but I can add them if you prefer.

**Comment by aingerson** [comment]
@wzamazon Thanks for the quick fix! I added the patch and it passes AWS. Failing Intel CI for unrelated cluster issues. It already passed our CI without the patch so I'm confident the extra efa patch shouldn't affect the results.

File: prov/util/src/util_attr.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (j-xiong) [reply prov/util/src/util_attr.c:1131]: That makes sense. No change is needed then.

[the remark this answers is not in the document]
Comment (shefty) [reply prov/util/src/util_attr.c:1131]: If we have hints, then the hint_caps are only what the application has requested.  So we use those as the minimum set of caps and OR in the secondary ones that are supported.  If we don't have hints, then we're reporting whatever the provider supports.  And as Alexia points out, the hint_caps are a subset of the others, with the check done prior to use even reaching this point.

**Review by shefty — APPROVED** [review]

**Pull request merged**